x/ccv/consumer/keeper/relay.go

```diff
@@ -223,6 +236,40 @@ func (k Keeper) SendPackets(ctx sdk.Context) {
 // in conjunction with the ibc module's execution of "acknowledgePacket",
 // according to https://github.com/cosmos/ibc/tree/main/spec/core/ics-004-channel-and-packet-semantics#processing-acknowledgements
 func (k Keeper) OnAcknowledgementPacket(ctx sdk.Context, packet channeltypes.Packet, ack channeltypes.Acknowledgement) error {
+	if res := ack.GetResult(); res != nil {
+		if len(res) != 1 {
+			k.Logger(ctx).Error("recv invalid ack; expected length 1", "channel", packet.SourceChannel, "ack", res)
+		}
+
+		// Unmarshal into V1 consumer packet data type. We trust data is formed correctly
+		// as it was originally marshalled by this module, and consumers must trust the provider
+		// did not tamper with the data. Note ConsumerPacketData.GetBytes() always JSON marshals to the
+		// ConsumerPacketDataV1 type which is sent over the wire.
+		var consumerPacket ccv.ConsumerPacketDataV1
+		ccv.ModuleCdc.MustUnmarshalJSON(packet.GetData(), &consumerPacket)
+		// If this ack is regarding a provider handling a vsc matured packet, there's nothing to do.
+		// As vsc matured packets are popped from the consumer pending packets queue on send.
+		if consumerPacket.Type == ccv.VscMaturedPacket {
+			return nil
+		}
+
+		// Otherwise we handle the result of the slash packet acknowledgement.
+		switch res[0] {
+		// We treat a v1 result as the provider successfully queuing the slash packet w/o need for retry.
+		case ccv.V1Result[0]:
+			k.ClearSlashRecord(ctx)           // Clears slash record state, unblocks sending of pending packets.
+			k.DeleteHeadOfPendingPackets(ctx) // Remove slash from head of queue. It's been handled.
+		case ccv.SlashPacketHandledResult[0]:
```

Comment:
It seems that the `case ccv.SlashPacketHandledResult[0]:` and the `case ccv.V1Result[0]:` are handled in the same way. Could that potentially mean that we could get rid of `SlashPacketHandledResult = PacketAckResult([]byte{byte(2)})` in `ccv.go`?

Reply:
See https://github.com/cosmos/interchain-security/pull/1024#discussion_r1286268187, but after revisiting this issue, yes we could potentially consolidate `SlashPacketHandledResult` and `V1Result` into a single enum value, as long as that value is 0x1. However, that enum value has completely different meaning once we implement provider changes. Seems like bad practice to change the meaning of an existing enum sent over the wire. 

pro of current design:
- Not overloading multiple meanings to an enum sent over the wire

con of current design:
- slightly more code

Reply:
>  Seems like bad practice to change the meaning of an existing enum sent over the wire.
Agreed! 👍